File: 50-Models/Elysio.Models/DTOs/AgentDTO.cs
namespace Elysio.Models.DTOs;

public record AgentDTO
{
    public Guid Id { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public string Name { get; set; } = string.Empty;
    public string Prompt { get; set; } = string.Empty;
    public double Temperature { get; set; }
    public string Model { get; set; } = string.Empty;

    public string? Description { get; set; }
    public List<RoomDTO> Rooms { get; set; } = [];
    public List<MessageDTO> Messages { get; set; } = [];
}

File: 10-Client/elysio.crew.front/src/models/interfaces/AgentDTO.ts
import type { RoomDTO } from "../interfaces/RoomDTO";
import type { MessageDTO } from "../interfaces/MessageDTO";

export type AgentDTO = {
  id: string;
  createdAt: string;
  updatedAt: string;
  name: string;
  prompt: string;
  temperature: number;
  model: string;
  description?: string;
  rooms: RoomDTO[];
  messages: MessageDTO[];
};
